### PR TITLE
tests: Enable soak parallel stability test

### PR DESCRIPTION
**Pull request by GabyCT**
This PR enables the soak parallel stability test for the kata GHA CI.

### CODE REVIEW

File: tests/stability/soak_parallel_rm.sh [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+count_containers() {
+	sudo ctr c list -q | wc -l
+}
+
+check_all_running() {
+	local goterror=0
+
+	echo "Checking ${how_many} containers have all relevant components"
```
Comment (dborquez) [inline tests/stability/soak_parallel_rm.sh:62]: If that is possible, use `info` helper function to print pretty console messages

File: tests/stability/soak_parallel_rm.sh [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+# we can also check it for sanity directly
+VC_POD_DIR="${VC_POD_DIR:-/run/vc/sbs}"
+
+# let's cap the test. If you want to run until you hit the memory limit
+# then just set this to a very large number
+MAX_CONTAINERS="${MAX_CONTAINERS:-110}"
+
+KATA_HYPERVISOR="${KATA_HYPERVISOR:-qemu}"
+
+check_vsock_active() {
```
Comment (dborquez) [inline tests/stability/soak_parallel_rm.sh:45]: to follow the uniformity used in the `tests` set, you could use the keyword `function` to define a function :)

**Review by dborquez — CHANGES_REQUESTED** [review]
Thank you @GabyCT, I'm leaving few small suggestions there.

**Comment by GabyCT** [comment]
@dborquez changes applied thanks

File: tests/stability/gha-run.sh [hunk 3]
```diff
@@ -13,16 +13,38 @@ kata_tarball_dir="${2:-kata-artifacts}"
...
+
+	sudo apt-get update
+	sudo apt-get -y install "${system_deps[@]}"
+
+	ensure_yq
+
+	declare -a github_deps
+	github_deps[0]="cri_containerd:$(get_from_kata_deps "externals.containerd.${CONTAINERD_VERSION}")"
+
+	for github_dep in "${github_deps[@]}"; do
```
Comment (chavafg) [inline tests/stability/gha-run.sh:31]: seems this loop is not needed as it is only one github dependency

**Review by dborquez — APPROVED** [review]
Thanks @GabyCT

File: tests/stability/soak_parallel_rm.sh [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+function count_containers() {
+	sudo ctr c list -q | wc -l
+}
+
+function check_all_running() {
+	local goterror=0
+
+	info "Checking ${how_many} containers have all relevant components"
+
+	# check what docker thinks
```
Comment (chavafg) [inline tests/stability/soak_parallel_rm.sh:64]: s/docker/ctr

File: tests/stability/soak_parallel_rm.sh [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+	if (( ${how_many_running} != ${how_many} )); then
+		info "Wrong number of containers running (${how_many_running} != ${how_many}) - stopping"
+		((goterror++))
+	fi
+
+	# Only check for Kata components if we are using a Kata runtime
+	if (( $check_kata_components )); then
+
+		# check we have the right number of shims
+		how_many_shims=$(pgrep -a -f ${SHIM_PATH} | grep containerd.sock | wc -l)
```
Comment (chavafg) [inline tests/stability/soak_parallel_rm.sh:76]: where does `SHIM_PATH` gets declared?

File: tests/stability/soak_parallel_rm.sh [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+	fi
+
+	versions_file="${cidir}/../versions.yaml"
+	nginx_version=$("${GOPATH}/bin/yq" read "$versions_file" "docker_images.nginx.version")
+	nginx_image="docker.io/library/nginx:$nginx_version"
+
+	# Pull nginx image
+	sudo ctr image pull ${nginx_image}
+	if [ $? != 0 ]; then
+		die "Unable to retry docker image ${nginx_image}"
```
Comment (chavafg) [inline tests/stability/soak_parallel_rm.sh:182]: s/retry/retrieve/?

**Review by amshinde — APPROVED** [review]

**Pull request merged**